esmvalcore/preprocessor/_area.py

```diff
@@ -177,13 +177,18 @@ def tile_grid_areas(cube, fx_files):
             fx_cube = iris.load_cube(fx_file)
 
             grid_areas = fx_cube.core_data()
+
             if cube.ndim == 4 and grid_areas.ndim == 2:
-                grid_areas = da.tile(grid_areas,
-                                     [cube.shape[0], cube.shape[1], 1, 1])
+                tiled = da.array([grid_areas for i in range(cube.shape[1])])
+                grid_areas = da.stack(tiled, axis=0)
+                tiled = da.array([grid_areas for i in range(cube.shape[0])])
+                grid_areas = da.stack(tiled, axis=0)
             elif cube.ndim == 4 and grid_areas.ndim == 3:
-                grid_areas = da.tile(grid_areas, [cube.shape[0], 1, 1, 1])
+                tiled = da.array([grid_areas for i in range(cube.shape[0])])
+                grid_areas = da.stack(tiled, axis=0).compute()
             elif cube.ndim == 3 and grid_areas.ndim == 2:
-                grid_areas = da.tile(grid_areas, [cube.shape[0], 1, 1])
+                tiled = da.array([grid_areas for i in range(cube.shape[0])])
+                grid_areas = da.stack(tiled, axis=0).compute()
```

Comment:
replace the two `elif`'s with:
```
elif (cube.ndim, grid_areas.ndim) in [(4, 3), (3, 2)]:
```
and write the defs for `tiled` and `grid_areas` only once